### PR TITLE
feature/se-48 & se-52

### LINKED ISSUE #48: Company Course Management - Front-end Creation
image

### PR DESCRIPTION
**Pull request by Ara-Ara-ASid-kun**
Closes https://github.com/SE2-WS23/documentation/issues/48 & https://github.com/SE2-WS23/documentation/issues/52

Added categories in Company Course Management Page, and added categories & training number in Company Trainings Overview page as per request.

Also added a few more examples for both pages

To add to that, used prettier to format all code as per request.

Of course, if you don't mind
Feet Fetish Meme

### CODE REVIEW

File: src/pages/CompanyCourseManagement/CompanyCourseManagementPage.jsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            display: 'flex',
+            justifyContent: 'center',
+          }}
+        >
+          {mockTrainings.map((training) => (
+            <TrainingListItem
+              key={training.trainingName}
+              trainingName={training.trainingName}
+              trainingCategory={training.trainingCategory}
+              backgroundColor={theme.palette.primary.main}
```
Comment (Arkcorner) [inline src/pages/CompanyCourseManagement/CompanyCourseManagementPage.jsx]: use uuidv4 for the Key you can orient yourself on the Quiz page

File: src/pages/CompanyTrainingOverview/CompanyTrainingsOverviewPage.jsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+          }}
+        >
+          {mockCourses.map((course) => (
+            <CourseListItem
+              key={course.courseName}
+              courseName={course.courseName}
+              courseCategory={course.courseCategory}
+              trainingNumber={course.trainingNumber}
+              backgroundColor={theme.palette.primary.main}
+            />
```
Comment (Arkcorner) [inline src/pages/CompanyTrainingOverview/CompanyTrainingsOverviewPage.jsx]: use uuidv4 as comment above

File: src/pages/CompanyCourseManagement/CompanyCourseManagementPage.jsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      >
+        <Button
+          variant='contained'
+          color='primary'
+          size='large'
+          sx={{ marginTop: '16px', alignSelf: 'flex-start' }}
+          onClick={handleAddTrainingClick}
+        >
+          ADD TRAINING
+        </Button>
```
Comment (Markus799) [inline src/pages/CompanyCourseManagement/CompanyCourseManagementPage.jsx]: The button is behind the navBar:
image

File: src/pages/CompanyCourseManagement/CompanyCourseManagementPage.jsx (hunk 1 above)
Comment (Ara-Ara-ASid-kun) [reply src/pages/CompanyCourseManagement/CompanyCourseManagementPage.jsx]: Requested changes have been implemented. Thank you

File: src/pages/CompanyTrainingOverview/CompanyTrainingsOverviewPage.jsx (hunk 2 above)
Comment (Ara-Ara-ASid-kun) [reply src/pages/CompanyTrainingOverview/CompanyTrainingsOverviewPage.jsx]: As per previous request, changes have also been implemented.

File: src/pages/CompanyCourseManagement/CompanyCourseManagementPage.jsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { Button, Grid } from '@mui/material';
+import Box from '@mui/material/Box';
+import PageLayout from '../../components/PageLayout';
+import { useTheme } from '@mui/material/styles';
+import { v4 as uuidv4 } from 'uuid';
+import mockTrainings from './mockTrainings';
+import TrainingListItem from './components/TrainingListItem';
+
+function CompanyCourseManagementPage() {
```
Comment (bishop6232) [inline src/pages/CompanyCourseManagement/CompanyCourseManagementPage.jsx]: please document your pages

File: src/pages/CompanyTrainingOverview/CompanyTrainingsOverviewPage.jsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { Button, Grid } from '@mui/material';
+import Box from '@mui/material/Box';
+import PageLayout from '../../components/PageLayout';
+import { useTheme } from '@mui/material/styles';
+import { v4 as uuidv4 } from 'uuid';
+import mockCourses from './mockCourses';
+import CourseListItem from './components/CourseListItem';
+
+function CompanyTrainingsOverviewPage() {
```
Comment (bishop6232) [inline src/pages/CompanyTrainingOverview/CompanyTrainingsOverviewPage.jsx]: add documentation too

**Review by bishop6232 — CHANGES_REQUESTED** [review]

File: src/pages/CompanyTrainingOverview/CompanyTrainingsOverviewPage.jsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+      >
+        <Button
+          variant='contained'
+          color='primary'
+          size='large'
+          sx={{ marginTop: '16px', alignSelf: 'flex-start' }}
+          onClick={handleAddCourseClick}
+        >
+          ADD COURSE
+        </Button>
```
Comment (Markus799) [inline src/pages/CompanyTrainingOverview/CompanyTrainingsOverviewPage.jsx]: The button is behind the navBar

File: src/pages/CompanyCourseManagement/CompanyCourseManagementPage.jsx (hunk 4 above)
Comment (Ara-Ara-ASid-kun) [reply src/pages/CompanyCourseManagement/CompanyCourseManagementPage.jsx]: Added documentation as requested.

File: src/pages/CompanyTrainingOverview/CompanyTrainingsOverviewPage.jsx (hunk 5 above)
Comment (Ara-Ara-ASid-kun) [reply src/pages/CompanyTrainingOverview/CompanyTrainingsOverviewPage.jsx]: Also added documentation as requested

File: src/pages/CompanyCourseManagement/CompanyCourseManagementPage.jsx [hunk 7]
```diff
@@ -0,0 +1,68 @@
+import React from 'react';
```
Comment (Markus799) [inline src/pages/CompanyCourseManagement/CompanyCourseManagementPage.jsx]: ``
'React' is defined but never used.

File: src/pages/CompanyCourseManagement/components/TrainingListItem.jsx [hunk 8]
```diff
@@ -0,0 +1,67 @@
+import React from 'react';
```
Comment (Markus799) [inline src/pages/CompanyCourseManagement/components/TrainingListItem.jsx]: 'React' is defined but never used.

File: src/pages/CompanyTrainingOverview/CompanyTrainingsOverviewPage.jsx [hunk 9]
```diff
@@ -0,0 +1,69 @@
+import React from 'react';
```
Comment (Markus799) [inline src/pages/CompanyTrainingOverview/CompanyTrainingsOverviewPage.jsx]: 'React' is defined but never used.

File: src/pages/CompanyTrainingOverview/components/CourseListItem.jsx [hunk 10]
```diff
@@ -0,0 +1,87 @@
+import React from 'react';
```
Comment (Markus799) [inline src/pages/CompanyTrainingOverview/components/CourseListItem.jsx]: 'React' is defined but never used.

File: src/pages/CompanyTrainingOverview/components/CourseListItem.jsx [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+
+CourseListItem.propTypes = {
+  courseName: PropTypes.string.isRequired,
+  courseCategory: PropTypes.string.isRequired,
+  trainingNumber: PropTypes.number.isRequired,
+  backgroundColor: PropTypes.string,
+};
+
+CourseListItem.defaultProps = {
+  backgroundColor: '#2196F3',
```
Comment (Markus799) [inline src/pages/CompanyTrainingOverview/components/CourseListItem.jsx]: > `backgroundColor: 'primary',`

File: src/pages/CompanyCourseManagement/components/TrainingListItem.jsx [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+TrainingListItem.propTypes = {
+  trainingName: PropTypes.string.isRequired,
+  trainingCategory: PropTypes.string.isRequired,
+  backgroundColor: PropTypes.string,
+};
+
+TrainingListItem.defaultProps = {
+  backgroundColor: '#2196F3',
```
Comment (Markus799) [inline src/pages/CompanyCourseManagement/components/TrainingListItem.jsx]: > `backgroundColor: 'primary',`

File: src/pages/CompanyCourseManagement/CompanyCourseManagementPage.jsx (hunk 3 above)
Comment (Ara-Ara-ASid-kun) [reply src/pages/CompanyCourseManagement/CompanyCourseManagementPage.jsx]: Fixed positioning of button for scrolling feature

File: src/pages/CompanyTrainingOverview/CompanyTrainingsOverviewPage.jsx (hunk 6 above)
Comment (Ara-Ara-ASid-kun) [reply src/pages/CompanyTrainingOverview/CompanyTrainingsOverviewPage.jsx]: Fixed positioning of button for scrolling feature

File: src/pages/CompanyCourseManagement/CompanyCourseManagementPage.jsx (hunk 7 above)
Comment (Ara-Ara-ASid-kun) [reply src/pages/CompanyCourseManagement/CompanyCourseManagementPage.jsx]: Removed react code

File: src/pages/CompanyTrainingOverview/CompanyTrainingsOverviewPage.jsx (hunk 9 above)
Comment (Ara-Ara-ASid-kun) [reply src/pages/CompanyTrainingOverview/CompanyTrainingsOverviewPage.jsx]: Removed react code

File: src/pages/CompanyTrainingOverview/components/CourseListItem.jsx (hunk 11 above)
Comment (Ara-Ara-ASid-kun) [reply src/pages/CompanyTrainingOverview/components/CourseListItem.jsx]: Removed default props after receiving feedback, deeming it unnecessary

File: src/pages/CompanyCourseManagement/components/TrainingListItem.jsx (hunk 12 above)
Comment (Ara-Ara-ASid-kun) [reply src/pages/CompanyCourseManagement/components/TrainingListItem.jsx]: Removed default props after receiving feedback, deeming it unnecessary

File: src/pages/CompanyTrainingOverview/components/CourseListItem.jsx (hunk 10 above)
Comment (Ara-Ara-ASid-kun) [reply src/pages/CompanyTrainingOverview/components/CourseListItem.jsx]: Removed react code

File: src/pages/CompanyCourseManagement/components/TrainingListItem.jsx (hunk 8 above)
Comment (Ara-Ara-ASid-kun) [reply src/pages/CompanyCourseManagement/components/TrainingListItem.jsx]: Removed react code

File: src/pages/CompanyCourseManagement/components/TrainingListItem.jsx [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+        border: '1px solid #ccc',
+        borderRadius: '8px',
+        backgroundColor,
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'flex-start',
+        justifyContent: 'flex-start',
+        margin: '10px',
+        padding: '10px',
+      }}
```
Comment (Markus799) [inline src/pages/CompanyCourseManagement/components/TrainingListItem.jsx:26]: Please use ListItem, have a look on the job-profile page

File: src/pages/CompanyTrainingOverview/components/CourseListItem.jsx [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+        borderRadius: '8px',
+        backgroundColor,
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'flex-start',
+        justifyContent: 'flex-start',
+        margin: '10px',
+        padding: '10px',
+        position: 'relative',
+      }}
```
Comment (Markus799) [inline src/pages/CompanyTrainingOverview/components/CourseListItem.jsx]: Please use ListItem, have a look on the job-profile page

**Pull request closed without merging**